shared-test/src/main/java/uk/gov/di/authentication/sharedtest/extensions/S3Extension.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.di.authentication.sharedtest.extensions;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.iterable.S3Objects;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import com.amazonaws.services.s3.model.HeadBucketRequest;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.net.URISyntaxException;
+
+public abstract class S3Extension extends BaseAwsResourceExtension
+        implements BeforeAllCallback, AfterAllCallback {
+
+    protected AmazonS3 s3Client;
+    protected static final String S3_ENDPOINT =
+            System.getenv().getOrDefault("S3_ENDPOINT", "http://localhost:45678");
+
+    @Override
+    public void beforeAll(ExtensionContext context) throws Exception {
+        var mockS3Credentials = new BasicAWSCredentials("access", "secret");
```

Comment:
You could mock the creds setup the client in the extension constructor.

Reply:
Discussed verbally - whole client will become a static member